management/networksecuritygroup/entities.go

```diff
@@ -0,0 +1,45 @@
+// Package networksecuritygroup implements operations for managing network security groups
+// using the Service Management REST API
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913824.aspx
+package networksecuritygroup
+
+import (
+	"encoding/xml"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+)
+
+// SecurityGroupClient is used to manage operations on network security groups
+type SecurityGroupClient struct {
+	client management.Client
+}
+
+// SecurityGroup represents a network security group
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913821.aspx
+type SecurityGroup struct {
```

Comment:
Although `struct SecurityGroup` looks the same in the "Create" request and "List" or "Get" response, they probably should be separate. In order to achieve that you can do:

```go
type SecurityGroup struct { ... }
type SecurityGroupCreateParameters SecurityGroup
type SecurityGroupResponse SecurityGroup
type SecurityGroupListResponse { SecurityGroups []SecurityGroupResponse }
```

I prefer to follow this convention (esp with SecurityGroupListResponse) because sometimes our  ListXxx API responses return more info than just an array of things (like they include a bit more info) and it's impossible to return those as well without breaking others.